release.py

```diff
@@ -0,0 +1,488 @@
+        release documents. It checks that the requested release tag is an acceptable bump
+        from the latest release tag.
+        """
+        latest_tag = self._get_latest_tag()
+        requested_release_tag = self.get_requested_release_tag()
+        self.release_validator.check_tag(latest_tag, requested_release_tag)
+        return requested_release_tag
+
+
+class ReleaseValidator:
+    def __init__(self, directory):
+        self.directory = directory
+
+    def check_repo(self):
+        if subprocess.check_output(['git', 'status', '--porcelain'], cwd=self.directory).strip():
+            exit('Repository is not clean. Commit or stash all changes')
+        else:
+            print 'Repository is clean'
+
+    @staticmethod
+    def check_branch_remote_exists(local_branch_name, remote_branches):
+        for remote_branch in remote_branches:
+            if local_branch_name == remote_branch['name']:
+                print 'Local branch %s exists remotely' % local_branch_name
+                return
+        exit('Local branch %s does not exist remotely' % local_branch_name)
+
+    @staticmethod
+    def check_tag(latest_tag, requested_release_tag):
+        print 'The latest release tag is %s.\n' \
+              'Detected requested release tag: %s' \
+              % (latest_tag, requested_release_tag)
+
+        latest_version_helper = SemanticVersionHelper(latest_tag)
+        acceptable_tags = latest_version_helper.get_acceptable_version_bumps()
+        if requested_release_tag not in acceptable_tags:
+            exit('Detected release tag %s is not part of the acceptable release tags: %s'
+                 % (requested_release_tag, acceptable_tags))
+
+    @staticmethod
+    def check_commit(last_local_commit, last_remote_commit):
+        if last_remote_commit != last_local_commit:
+            exit('Last local and remote commits do not match')
+        else:
+            print 'Last local and remote commits match'
+
+    @staticmethod
+    def _get_and_check_release_file(file_path, string_contained=None, string_begins=None):
+        with open(file_path, 'r') as release_file:
+            file_contents = release_file.read()
+            if string_contained:
+                if string_contained not in file_contents:
+                    exit('Expected "%s" to be in %s' % (string_contained, file_path))
+            if string_begins:
+                if not file_contents.startswith(string_begins):
+                    print file_contents
+                    exit('Expected %s to begin with "%s"' % (file_path, string_contained))
+
+            return file_contents
+
+    def _confirm_version_changed_in_file(self, file_path, tag_name):
+        contents = self._get_and_check_release_file(file_path)
+        for line in contents.splitlines():
+            if 'version' in line and '=' in line:
+                if tag_name not in line:
+                    exit('Version has not been updated to %s in %s' % (tag_name, file_path))
```

Comment:
Actually, it would probably be preferable to extract a version constant in setup.py, and then `from prestoadmin import version` and `from setup import version`, which lets us rely on the python parser.